contracts/utils/Arrays.sol

```diff
@@ -20,6 +20,9 @@ library Arrays {
      *
      * `array` is expected to be sorted in ascending order, and to contain no
      * repeated elements.
+     *
+     * Deprecated in favor of `lowerBound` and `findUpperBound`. Note that this actually implements a lower bound
```

Comment:
I can't make a suggestion, but we should change the NatSpec of this function to reflect its actual behavior. Right now it's saying it does one thing and it "notes" that it does another.

What about?

```
* @dev Searches a sorted `array` and returns the last index that contains
* a value greater or equal to `element`. If no such index exists (i.e. all
* values in the array are strictly less than `element`), the array length is
* returned. Time complexity O(log n).
```

And then this note:

```suggestion
     * IMPORTANT: Deprecated. This implementation behaves as {lowerBound}, which should be used instead.
```